modules/http1/http1_core.c

```diff
@@ -0,0 +1,142 @@
+/* Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "apr_strings.h"
+#include "apr_thread_proc.h"    /* for RLIMIT stuff */
+
+#define APR_WANT_STRFUNC
+#include "apr_want.h"
+
+#include "httpd.h"
+#include "http_core.h"
+#include "http_log.h"
+#include "http_protocol.h"   /* For index_of_response().  Grump. */
+#include "http_request.h"
+
+#include "util_filter.h"
+#include "util_ebcdic.h"
+#include "ap_mpm.h"
+#include "scoreboard.h"
+
+#include "mod_core.h"
+#include "mod_http1.h"
+
+/* Handles for core http/1.x filters */
+AP_DECLARE_DATA ap_filter_rec_t *ap_chunk_filter_handle;
+AP_DECLARE_DATA ap_filter_rec_t *ap_http1_response_out_filter_handle;
+AP_DECLARE_DATA ap_filter_rec_t *http1_request_in_filter_handle;
+AP_DECLARE_DATA ap_filter_rec_t *http1_body_in_filter_handle;
+
+
+static void http1_pre_read_request(request_rec *r, conn_rec *c)
+{
+    if (!r->main && !r->prev
+        && !strcmp(AP_PROTOCOL_HTTP1, ap_get_protocol(c))) {
+        if (r->proxyreq == PROXYREQ_NONE) {
+            ap_add_input_filter_handle(http1_request_in_filter_handle,
+                                       NULL, r, r->connection);
+        }
+        ap_add_output_filter_handle(ap_http1_response_out_filter_handle,
+                                    NULL, r, r->connection);
+    }
+}
+
+static int http1_post_read_request_early(request_rec *r)
+{
+    if (!r->main && !r->prev
+        && !strcmp(AP_PROTOCOL_HTTP1, ap_get_protocol(r->connection))) {
+        /* HTTP1_BODY_IN takes care of chunked encoding and content-length.
+         */
+        ap_add_input_filter_handle(http1_body_in_filter_handle,
+                                   NULL, r, r->connection);
+    }
+    return OK;
+}
+
+static int http1_post_read_request(request_rec *r)
+{
+    const char *tenc;
+
+    if (!r->main && !r->prev
+        && !strcmp(AP_PROTOCOL_HTTP1, ap_get_protocol(r->connection))) {
+        if (r->proto_num >= HTTP_VERSION(1,0)) {
```

Comment:
r->proto_num >= HTTP_VERSION(1,1) (instead of strcmp + proto_num checks)?

Reply:
Implemented.